mne/preprocessing/infomax_.py

```diff
@@ -169,7 +177,8 @@ def infomax(data, weights=None, l_rate=None, block=None, w_change=1e-12,
                 # extended ICA update
                 y = np.tanh(u)
                 weights += l_rate * np.dot(weights,
-                                           BI - np.dot(np.dot(u.T, y), signs) -
+                                           BI -
+                                           signs[None, :] * np.dot(u.T, y) -
```

Comment:
FYI you could avoid the `None` here by making your arrays `ones((1, n_features))` above, but it would necessitate changes elsewhere. But you should use `np.newaxis` instead of `None`. Even though right now it's an alias, it's a bit clearer and if they change it to something else in the future then we're future compatible.